projects/ng-draw-flow/src/lib/directives/selectable-element/selectable-element.directive.ts

```diff
@@ -11,19 +11,55 @@ import {
 export class SelectableElementDirective {
     private selected = false;
     private readonly el = inject(ElementRef);
+    private isDragging = false;
+    private startX: number | null = null;
+    private startY: number | null = null;
+    private readonly dragThreshold = 5; // Displacement threshold for drag detection
 
     @Output()
     protected readonly selectionChanged = new EventEmitter<boolean>();
 
-    @HostListener('document:mousedown', ['$event.target'])
-    protected onDocumentClick(targetElement: any): void {
-        const clickedInside = this.el.nativeElement.contains(targetElement);
+    @HostListener('document:mousedown', ['$event'])
+    protected onMouseDown(event: MouseEvent): void {
+        this.startX = event.clientX;
+        this.startY = event.clientY;
+        const targetElement = event.target as HTMLElement;
+
+        this.isDragging = false;
+
+        if (targetElement.dataset.element !== 'scene') {
+            const clickedInside = this.el.nativeElement.contains(targetElement);
+
+            this.setSelected(clickedInside);
+        }
+    }
+
+    @HostListener('document:mousemove', ['$event'])
```

Comment:
nit: according to angular doc 

> Always prefer using the host property over @HostBinding and @HostListener. These decorators exist exclusively for backwards compatibility.

(we in the taiga have already completely moved to this approach)